### DISCUSSION TITLE
Fleet Node web interface timing out - so-firewall returning FileNotFoundError

### DISCUSSION THREAD

**Discussion by mskarshinski** [Security-Onion-Solutions/securityonion#7743] [Unsupported Versions] [answered]
I have a distributed setup with a Manager, a Search node, a Suricata forwarder, and now a dedicated Fleet forwarder.  Fleet has been kicking me in the balls for two days.  Fleet was *initially* installed on the Manager (I read another post that implied that this could require some special attention)

I *think* it is the firewall on the Fleet node that is the issue, but my attempts to use so-firewall are failing:

[mskar@wst-norf-sos-sn2 ~]$ sudo so-firewall listhostgroups
Traceback (most recent call last):
  File "/sbin/so-firewall", line 392, in <module>
    main()
  File "/sbin/so-firewall", line 382, in main
    code = cmd(options, args[1:])
  File "/sbin/so-firewall", line 285, in listhostgroups
    return listHostGroups()
  File "/sbin/so-firewall", line 81, in listHostGroups
    content = loadYaml(hostgroupsFilename)
  File "/sbin/so-firewall", line 73, in loadYaml
    file = open(filename, "r")
FileNotFoundError: [Errno 2] No such file or directory: '/opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml'

When I go poking around for files, it appears to me that /opt/so /nsm and /etc are VERY sparse on the Fleet node.
[mskar@wst-norf-sos-sn2 ~]$ sudo ls -R /opt/so/saltstack
/opt/so/saltstack:
(nothing found)

Any help would be GREATLY appreciated, I'm spinning my wheels at this point.  

Thanks!
Marc

**Comment by defensivedepth** [comment]
hey there Marc - 

There should not need to modify the firewall manually on the standalone FleetDM node. Installing FleetDM on the Manager initially and then setting up a dedicated Fleet node is a supported process and should work without any extra config. The older version of FleetDM on the Manager will automatically deprecate itself and point to the new dedicated Fleet node.

When you login to the Security Onion Console and click on the FleetDM link, does it bring up the FleetDM interface?

**Reply by mskarshinski** [reply]
Good to know, I saw several references to running so-firewall and was concerned that I could not do so from this node.  A couple of relevant things to point out:
1. I was unable to complete any of my sensor node setups without changing my hosts file midway through the setup to point the actual address at the sensor name/fqdn, rather than 127.0.0.1 (I left localhost entries at 127.0.0.1 of course).  Without that manual change, the setup would fail after 95% with a generic error.
2. Once I got my fleet node setup, clicking on fleet in my SOC attempted to send me to the IP rather than hostname, I was able to change this in the config file.
3. My managerssl.crt and managerssl.key from the manager were not copied into the sensor's /etc/pki or /etc/pki/nginx folder, in fact, that folder was empty. I use a wildcard certificate for my web interfaces and I had to copy this into the /etc/pki folder and then create the relevenat links in the /etc/pki/nginx folder.  I am completely unclear on how salt is supposed to distribute, from where, and to where.  Is there a document somewhere I'm missing?
4. Once I got my certificates setup, and my config modified, I can now click on fleet and it brings me to the web server on my fleet node, with the correct/valid certficate.  However, none of my Nodes are checking in any longer (currently just my SO boxes).  If I create a deploy package, it doesn't seem to be able to register.

I'm happy to supply relevant config/log files, just unsure of which to provide.  I know this is a lot, I really appreciate the help!

**Reply by defensivedepth** [reply]
Is everything else working correctly? (Suricata generating alerts, able to view and escalate in the alerts console, etc?)

Needing to change the hosts file midway through setup is indicative of a deeper issue.

**Reply by mskarshinski** [reply]
Everything is tip-top otherwise.  Suricata, Wazuh eventlog, and Playbook entries are all generating in my SOC.  I read the "modify the hosts file manually" in another post, and it did the trick for me.  I am likely diverging from the stock install in the following ways...
I *MUST* have valid TLS for my web interfaces, its policy.  I have accomplished this by putting my wildcard cert-chain and key into the /etc/pki folder on the manager, and have had to repeat this on the fleet node, which is the only other node that serves up an interface AFAIK.  I kind of sidled my way into figuring this out, and its possible that I'm putting it in the wrong place, or need to put it somewhere in the saltstack as well?
I'm not sure why the stock installation doesn't want a FQDN, instead using .localdomain, adding the true domainname to the search list but not to the hostname.  I did use hostctl -sethostname to my FQDN on at least one node, I believe it was my Wazuh sensor node.  Maybe having the domain set properly is inconsequential and I'm just perseverating on minutiae that does not matter (wouldn't be the first time).
I'm in a pretty reasonable place right now with everything except for Fleet.  All of my hosts are showing as registered, but offline.  I had to travel this week for work, but will be hitting this again this afternoon hopefully and will post additional information.  I know I'm not giving you all that much to go on so I'll add some notable logs and configs to this ticket.  I will also start reading forum posts NOT related to issues I'm trying to solve.  I am gaining a lot of knowledge about how SO works, internally, from the responses to other people's tickets

Thank you VERY much for your help, this product is a game-changer for me, its flexibility and ability to be customized is why I prefer Community/Open Source solutions overall (especially pfSense).  I know that a lot of the issues i'm having are due to my own limited understanding of the intracacies of Docker and Salt but I continue to research and build my understanding.

**Comment by mskarshinski** [comment, marked as answer]
You can close this ticket, I believe I have found what I was looking for.  Thanks for your help!

**Reply by defensivedepth** [reply]
@mskarshinski What ended up being the issue?

**Reply by mskarshinski** [reply]
I haven't really *fixed* anything yet, but I did see another post that discusses that the Fleet certificates need to be rebuilt after a host name change.  My last two posts to this thread lacked brevity and I feel like I was just confusing the issue.  I'm fine closing this and picking it up down the road.  Fleet is not critical for my installation, I have all of its functionality baked into my RMM (Syncro).  Thanks for reaching out!
